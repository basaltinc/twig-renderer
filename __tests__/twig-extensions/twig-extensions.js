const path = require('path');

const TwigRenderer = require('../../src');

describe('Twig Extensions', () => {
  const twigRenderer = new TwigRenderer({
    src: {
      roots: [path.join(__dirname, 'src')],
    },
    alterTwigEnv: [{
      file: path.join(__dirname, 'alter-twig.php'),
      functions: ['addCustomExtension'],
    }],
  });

  test('Twig Extensions - Simple Function', async () => {
    const results = await twigRenderer.render('extension-test.twig');

    expect(results.ok).toEqual(true);
    expect(results).toMatchSnapshot();
  });

  afterAll(() => twigRenderer.closeServer());
});
