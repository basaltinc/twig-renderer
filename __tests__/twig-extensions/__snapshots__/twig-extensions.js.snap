// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twig Extensions Twig Extensions - Simple Function 1`] = `
Object {
  "html": "<h1>Hello Custom World!</h1>",
  "ok": true,
}
`;
